### ISSUE TITLE
Clear() for cells sets the value to "", not null

### ISSUE THREAD

**Issue by kendallb** [ClosedXML/ClosedXML#544]
## Read and complete the full issue template

- [x] Bug
Version of ClosedXML: 0.89.0

**What is the current behavior?**

When you set a cell to null, the cell value is set to a blank string when you read it back. 

**What is the expected behavior or new feature?**

Excel generally stores empty cells as null values in the Excel files, but ClosedXML is unable to either read the null values as nulls nor write empty cells as nulls on writing.

I am in the process of re-writing our Excel reading/writing library (basically a port of CsvHelper to work with Excel) from the ComponentOne C1.Excel library to ClosedXML. One of the features we supported in our library was the ability to handle empty cells and provide default values when records are loaded, but for this to work it relies on actually reading an empty cell from the Excel file as a null. For our testing I need the ability to write empty cells as nulls so I can read them back that way as well. 

It would appear that an architectural decision was made to not support empty cells as null but rather as simply blank cells, but the problem with this approach is you have data loss because you cannot tell when you load an Excel file if the cell was actually empty, or if someone literally put a blank string in there? 

An empty cell and a blank cell are actually two different things.

**Did this work in previous versions of our tool?  Which versions?**

I doubt it.

**Code to reproduce problem:**
```c#
        [TestMethod]
        public void NullTest()
        {
            using (var stream = new MemoryStream()) {
                // Create some test data to parse
                using (var book = new XLWorkbook()) {
                    var sheet = book.AddWorksheet("Sheet 1");
                    sheet.Cell(1, 1).Value = "test1";
                    // Cell(1,2) is empty
                    sheet.Cell(1, 3).Value = null;
                    book.SaveAs(stream);
                }

                // Now load it back
                stream.Position = 0;
                using (var book = new XLWorkbook(stream)) {
                    var sheet = book.Worksheets.Worksheet(1);
                    var result =  sheet.Cell(1, 1).Value;
                    Assert.AreEqual("test1", result);
                    result = sheet.Cell(1, 2).Value;
                    Assert.AreEqual(null, result);  // Should be null!
                    result = sheet.Cell(1, 3).Value;
                    Assert.AreEqual(null, result);  // Should be null!
                }
            }
        }
```

**Comment by igitur** [comment]
Sorry, don't know how this issue slipped past me. 

I'm not really convinced by this. ClosedXML sometimes (not always) attempts to replicate the API provided by VBA. So, for a blank workbook, if I open the VBA editor and enter this in the `Immediate` window:
`? Sheet1.Cells(1,1).Value = ""`
I get `True`

I've never seen Excel treat a cell value as `null`. Changing this in ClosedXML would be a major change that would break many users' code, including my own.

cc @cobysolutions

**Comment by cobysolutions** [comment]
Gotcha.

**Comment by kendallb** [comment]
I believe when Excel *reads* a cell, a null is the same as a blank. But when it is written it would appear it can actually write both a null and a blank and know the difference? I have certainly read Excel files using the C1.Excel library where cells have null values in them, not blanks. I know that because I had to write code to specifically handle those nulls when I do type conversion and for our Excel reading code we have an option to convert null strings to blank.

**Comment by igitur** [comment]
Can you perhaps produce a file some way that shows the difference between a cell with a null value vs a cell with an empty string value?

**Comment by kendallb** [comment]
Ok here is a test file. I wrote the file with the first column as null, the second one as blank and the third with a string in it using the C1.Excel library. However when I read it back with C1.Excel, both the first two columns actually load as null? I wonder if C1.Excel has always been considering empty cells in an Excel file to be null and not a blank string? Probably saves memory that way, and I we had to write code to make sure null strings got converted into blanks for our parsing code as blanks as much easier to manager than nulls when parsing.

[test.xlsx](https://github.com/ClosedXML/ClosedXML/files/1718089/test.xlsx)
